code/modules/station_goals/bsa.dm

```diff
@@ -341,8 +352,7 @@ CREATION_TEST_IGNORE_SUBTYPES(/obj/machinery/power/bsa/full)
 
 	var/datum/weakref/cannon_ref
 	var/notice
-	var/target
-	var/area_aim = FALSE //should also show areas for targeting
+	var/datum/target
```

Comment:
This is better to use weak ref. hard del issue.

Reply:
done